### ISSUE TITLE
provider property values must be strings

### ISSUE THREAD

**Issue by jaredallard** [pulumi/pulumi-vault#27]
Hi! I'm attempting to use this. This _should_ repro it:

```
日本 [infrastructure] pulumi up
Previewing update (authz):
     Type                       Name         Plan     Info
     pulumi:pulumi:Stack        authz-authz
     └─ pulumi:providers:vault  vault                 3 errors

Diagnostics:
  pulumi:providers:vault (vault):
    error: pulumi:providers:vault resource 'vault's property 'maxRetries' value {2} has a problem: provider property values must be strings
    error: pulumi:providers:vault resource 'vault's property 'maxLeaseTtlSeconds' value {1200} has a problem: provider property values must be strings
    error: pulumi:providers:vault resource 'vault's property 'skipTlsVerify' value {false} has a problem: provider property values must be strings
```

Repro:

```go
		vaultProvider, err := vault.NewProvider(ctx, "vault", &vault.ProviderArgs{
			Address: pulumi.StringPtr("https://vault.outreach.cloud"),
		})
		if err != nil {
			return err
		}

		_, err = vaultGeneric.NewSecret(ctx, name, &vaultGeneric.SecretArgs{
			DataJson: pulumi.String("..."),
		}, pulumi.Provider(vaultProvider))
```

### Version

 * pulumi v1.9.1
 * pulumi-vault v1.8.0

**EDIT**: For anyone running into this... for now you can't pass in a custom provider, you'll have to leave it unconfigured and use EnvVars instead.

**Comment by jaredallard** [comment]
This is blocking a PoC to demonstrate the functionality of this to my org. I hate to double respond, does anyone have some insight? :pray:

**Comment by jaredallard** [comment]
Sorry to bump this again, any ideas on when this will be done @stack72? If not any time soon, some pointers on how to fix this would be awesome as well. Happy to implement this myself since I have a tight timeline.

**Comment by stack72** [comment]
Hi @jaredallard 

This is on my list to fix today

Paul

**Comment by lukehoban** [comment]
Did https://github.com/pulumi/pulumi/pull/3948 fix this?

**Comment by stack72** [comment]
@lukehoban looks like https://github.com/pulumi/pulumi/pull/3948 did indeed fix this - this will make the 1.11.1 release it seems :D
